### PR TITLE
[v7r1] Stop looking for SEs in BDII

### PR DESCRIPTION
**Pull request by fstagni**
For #3908

BEGINRELEASENOTES

*Configuration
CHANGE: BDII2CSAgent: stop looking for SEs in BDII

ENDRELEASENOTES

### CODE REVIEW

File: ConfigurationSystem/Client/Utilities.py [hunk 1]
```diff
@@ -463,6 +465,71 @@ def getGridSRMs(vo, bdiiInfo=None, srmBlackList=None, unUsed=False):
...
+  """ Get all the SRMs available for a given VO
+
+      :param str vo: VO name
+      :param dict bdiiInfo: information from BDII
+      :param list sesBlackList: SRMs from black list
+      :param bool unUsed: unused
+
+      :return: S_OK(dict)/S_ERROR()
+  """
+  result = ldapService(serviceType='???', vo=vo)
```
Comment (fstagni) [inline ConfigurationSystem/Client/Utilities.py:478]: @andresailer @atsareg what should I use here for getting all SEs types?

Comment (andresailer) [reply ConfigurationSystem/Client/Utilities.py:478]: For GLUE2 I would guess it is not that easy.

Comment (fstagni) [reply ConfigurationSystem/Client/Utilities.py:478]: Here for reference the GridPP code: https://github.com/ic-hep/GridPPDIRAC/blob/master/ConfigurationSystem/private/AutoBDIISEs.py

They seem to look also for `xroot` with:
```
    xrootports = ldap_conn.search_s(base=base, scope=scope,
                                    filterstr="(&(objectClass=GlueSEAccessProtocol)"
                                    "(GlueChunkKey=*)"
                                    "(GlueSEAccessProtocolEndpoint=*)"
                                    "(|(GlueSEAccessProtocolType=Root)"
                                    "(GlueSEAccessProtocolType=Xroot)))")
```

**Comment by fstagni** [comment]
I am tempted to simply remove the lookup for storages from BDII2CSAgent. In theory things are moving towards "SRR": https://twiki.cern.ch/twiki/bin/view/LCG/StorageSpaceAccounting

Convince me otherwise.

**Comment by fstagni** [comment]
Thanks @chaen for the pointer, in the last commit I factorized better the interactions between the 2.

**Comment by fstagni** [comment]
Review, please!

File: ConfigurationSystem/Client/Helpers/Resources.py [hunk 2]
```diff
@@ -310,53 +310,6 @@ def getQueues(siteList=None, ceList=None, ceTypeList=None, community=None, mode=
   return S_OK(resultDict)
 
 
-def getStorageElements(vo=None):
```
Comment (atsareg) [inline ConfigurationSystem/Client/Helpers/Resources.py:313]: Why this function is removed ?

File: Resources/scripts/dirac-resource-info.py [hunk 3]
```diff
@@ -101,17 +102,14 @@ def printCEInfo(voName):
...
-  if not result['OK']:
-    gLogger.error('Failed to get SE information')
-    DIRACExit(-1)
-  seDict = result['Value']
-
   fields = ('SE', 'Status', 'Protocols', 'Aliases')
   records = []
 
-  for se in seDict:
+  for se in DMSHelpers(voName).getStorageElements():  # this will get the full list of SEs, not only the vo's ones.
```
Comment (atsareg) [inline Resources/scripts/dirac-resource-info.py:108]: May be DMSHelpers.getStorageElements should better go to Resources ?

Comment (chaen) [reply Resources/scripts/dirac-resource-info.py:108]: The `DMSHelper` is nice because it is dynamic, and adapted to the running conditions (VO, local site, remove duplicate alias, etc). 
I see `Resources` as a more static set of tools: just list what is in `StorageElements` for example. And `DMSHelpers` could (should) use `Resources`. 
But now I am unsure what is being done here, and whether we need just the static list (`Resources) or a more filtered one (`DMSHelper`).  
Maybe after all I was wrong suggesting to @fstagni that there should be only one method `getStorageElements` ?

File: ConfigurationSystem/Client/Helpers/Resources.py (hunk 2 above)
Comment (fstagni) [reply ConfigurationSystem/Client/Helpers/Resources.py:313]: Because it was never used and there's `DMSHelper` function.

File: Resources/scripts/dirac-resource-info.py (hunk 3 above)
Comment (fstagni) [reply Resources/scripts/dirac-resource-info.py:108]: I can re-add a `Resources.getStorageElements` function like this:

```
def getStorageElements(vo):
  ses = []

  for se in DMSHelpers(voName).getStorageElements():
    seObject = StorageElement(se)
    if not (seObject.vo and voName in seObject.vo.strip().split(',') or not seObject.voName):
      continue
    ses.append(se)
```

Comment (chaen) [reply Resources/scripts/dirac-resource-info.py:108]: well, the question is: what is this supposed to do ? :-D

Comment (fstagni) [reply Resources/scripts/dirac-resource-info.py:108]: It will get the SEs per VO (useful for multi-VO setup)

Comment (chaen) [reply Resources/scripts/dirac-resource-info.py:108]: I am sorry if I created some confusions, but maybe now is a good time to sort it out ?
@phicharp wrote most of the DMSHelpers so he probably has some input to give.

Actually, as of now, `Resources.getStorageElements` does not work (or rather, will break in v7r1) because it expects to find the `BaseSE` definition in `StorageElements`, while it will be mandatory to have it in `StorageElementBases`. What it was supposed to do, from what I understand, is to return the list of SE together with the configuration options, and limited to a given VO.

The `DMSHelper` has a very convoluted way of exploring, since it actually explores  the list of Sites and the associated SE, before adding from `StorageElements` CS sections what was not already added. But it basically boils down to listing all the `StorageElements` CS sections, so not filtered by VO. 

I have checked and `Resources.getStorageElements` is used only by one script, so I think we could just ditch it for now. 

But we have kind of a circular dependency issue here, that would be good to solve.
* The `CS.Helpers` are just grouped/exploded CS sections that belong to the same sections (`Operations`, `Resources`, etc). 
* the `DMSHelper` uses several of these `CS.Helpers` to return information that makes sense from the DMS point of view. That's why I think the VO filtering should be done here. 
* The difficulty lies in the fact that to have something sensible, listing the `StorageElements` CS section is not enough. One needs to resolve the `Alias` and `BaseSE` mess, and this is quite complex, and already done properly in the `StorageFactory`. This `StorageFactory` (which is not a factory from the DP point of view, but who reads the Big Four books anyway...) is somewhat an internal object, used by the `StorageElement` to get its full config. But the `StorageElement` relies on `DMSHelper`, so the `DMSHelper` cannot just instantiate `StorageElements`. 

Bottom line: I do not have a clean a future proof solution right now, so I propose to create an issue for that, think about it for the future, and for the time being, just ditch the `Resources.getStorageElements` part. Bloody multi vo....
 @atsareg @andresailer opinions ?

Comment (fstagni) [reply Resources/scripts/dirac-resource-info.py:108]: This PR ditches `Resources.getStorageElements`, so I guess it's fine for the moment. Maybe temporarily. Can this PR be approved now?

Comment (atsareg) [reply Resources/scripts/dirac-resource-info.py:108]: OK for now

**Review by atsareg — APPROVED** [review]
